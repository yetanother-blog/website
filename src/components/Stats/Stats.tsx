import styled from 'styled-components';
import { margin, MarginProps } from 'styled-system';

export type StatsProps = MarginProps;

export const Stats = styled.div<StatsProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  ${margin}

  @media (min-width: ${(props) => props.theme.breakpoints[1]}) {
    flex-direction: row;
  }

  & > * {
    position: relative;
    padding-right: ${(props) => props.theme.space.xl}px;
    padding-left: ${(props) => props.theme.space.xl}px;
    margin-bottom: ${(props) => props.theme.space.xl}px;

    &:last-child {
      margin-bottom: 0;
    }

    @media (min-width: ${(props) => props.theme.breakpoints[1]}) {
      padding-right: ${(props) => props.theme.space.xxl}px;
      padding-left: ${(props) => props.theme.space.xxl}px;
      margin-bottom: 0;
    }
  }

  @media (min-width: ${(props) => props.theme.breakpoints[1]}) {
    & > *::after {
      content: '';
      width: 8px;
      height: 80%;
      position: absolute;
      right: -4px;
      top: 10%;
      background-color: ${(props) => props.theme.colors.grey300};
    }

    & > *:last-child::after {
      display: none;
    }
  }
`;
