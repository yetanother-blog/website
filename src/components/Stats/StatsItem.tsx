import React from 'react';
import { Link } from '../Link/Link';
import { Typography } from '../Typography/Typography';

export interface StatsItemPros {
  value: string;
  unit: string;
  description: string;
  source: string;
  sourceUrl: string;
}

export const StatsItem: React.FC<StatsItemPros> = ({
  value,
  unit,
  description,
  source,
  sourceUrl,
}) => {
  return (
    <div>
      <Typography
        variant="title"
        color="secondary"
        as="span"
        display="block"
        textAlign="center"
        marginBottom="s"
      >
        {value}
        <small>{unit}</small>
      </Typography>
      <Typography
        variant="smallText"
        as="span"
        display="block"
        textAlign="center"
        marginBottom="m"
      >
        {description}
      </Typography>
      <Typography
        variant="tinyText"
        textAlign="center"
        as="span"
        display="block"
      >
        (Source:&nbsp;
        <Link
          variant="quartenary"
          href={sourceUrl}
          target="_blank"
          rel="noopener noreferrer nofollow"
        >
          {source}
        </Link>
        )
      </Typography>
    </div>
  );
};
