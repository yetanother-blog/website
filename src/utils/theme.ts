import { DefaultTheme } from 'styled-components';

export const theme: DefaultTheme = {
  fontSizes: ['12px', '14px', '16px', '20px', '24px', '32px'],
  colors: {
    primary: '#00FF00',
    secondary: '#FB00FF',
    grey500: '#1C1C1C',
    grey400: '#ACADAC',
    grey300: '#F6F6F6',
  },
  breakpoints: ['576px', '768px', '992px', '1200px'],
  space: {
    zero: 0,
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 20,
    xl: 40,
    xxl: 60,
    xxxl: 120,
  },
  radii: {
    s: 3,
  },
  width: {
    m: '620px',
    l: '940px',
    xl: '1200px',
  },
};
