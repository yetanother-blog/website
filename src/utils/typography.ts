import { CSSObject } from "styled-components";
import Typography from "typography";

const typography = new Typography({
  googleFonts: [
    {
      name: "Source Sans Pro",
      styles: ["400", "600"]
    },
    {
      name: "Source Code Pro",
      styles: ["600", "700"]
    }
  ],
  baseFontSize: "20px",
  baseLineHeight: 1.85,
  bodyFontFamily: ["Source Sans Pro", "sans-serif"],
  boldWeight: "600",
  overrideStyles: ({ adjustFontSizeTo }, options, styles) => ({
    a: {
      fontFamily: `"Source Code Pro", "sans-serif"`
    }
  })
});

// Hot reload typography in development.
if (process.env.NODE_ENV !== `production`) {
  typography.injectStyles();
}

export default typography;

export const rhythm = typography.rhythm;
export const scale = typography.scale;

type StyledScale = (values: number) => CSSObject;
export const styledScale = scale as StyledScale;
