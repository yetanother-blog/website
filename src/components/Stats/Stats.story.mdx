import { Story, Meta, Preview, Props } from '@storybook/addon-docs/blocks';
import { Stats } from './Stats';
import { StatsItem } from './StatsItem';

<Meta title="Components/Stats" />

# Stats

Stats component to show some fancy data in blog posts.

<Preview>
  <Story name="default">
    <React.Fragment>
      <Stats>
        <StatsItem 
          value="~50" 
          unit="%" 
          description="Of all top 1m websites by traffic include Google Analytics" 
          source="BuiltWith" 
          sourceUrl="https://trends.builtwith.com/analytics/Google-Analytics" 
        />
        <StatsItem 
          value="~46" 
          unit="KB" 
          description="Size of the Google Analytics JavaScript Snippet" 
          source="Google" 
          sourceUrl="https://www.google-analytics.com/analytics.js" 
        />
      </Stats>
    </React.Fragment>
  </Story>
</Preview>

## Props

<Props of={Stats} />
