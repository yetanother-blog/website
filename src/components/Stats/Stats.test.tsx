import React from 'react';
import { render } from '@testing-library/react';
import { ThemeContext } from 'styled-components';
import { theme } from '../../utils/theme';
import { Stats } from './Stats';
import { StatsItem } from './StatsItem';

describe('Stats component', () => {
  it('should render', () => {
    const { container } = render(
      <ThemeContext.Provider value={theme}>
        <Stats>
          <StatsItem
            value="~50"
            unit="%"
            description="Of all top 1m websites by traffic include Google Analytics"
            source="BuiltWith"
            sourceUrl="https://trends.builtwith.com/analytics/Google-Analytics"
          />
          <StatsItem
            value="~46"
            unit="KB"
            description="Size of the Google Analytics JavaScript Snippet"
            source="Google"
            sourceUrl="https://www.google-analytics.com/analytics.js"
          />
        </Stats>
      </ThemeContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
