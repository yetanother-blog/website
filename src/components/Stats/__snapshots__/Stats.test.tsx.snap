// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats component should render 1`] = `
<div>
  <div
    class="sc-bdvvaa hITgeK"
  >
    <div>
      <span
        class="sc-gKckTs cBMlwg"
        color="secondary"
        display="block"
      >
        ~50
        <small>
          %
        </small>
      </span>
      <span
        class="sc-jrQzUz jxTuHj"
        display="block"
      >
        Of all top 1m websites by traffic include Google Analytics
      </span>
      <span
        class="sc-kDThTU cwQFAH"
        display="block"
      >
        (Source: 
        <a
          class="sc-jRQAMF hVqlct"
          href="https://trends.builtwith.com/analytics/Google-Analytics"
          rel="noopener noreferrer nofollow"
          target="_blank"
        >
          <span
            class="sc-gsDJrp leAAmy"
          >
            BuiltWith
          </span>
        </a>
        )
      </span>
    </div>
    <div>
      <span
        class="sc-gKckTs cBMlwg"
        color="secondary"
        display="block"
      >
        ~46
        <small>
          KB
        </small>
      </span>
      <span
        class="sc-jrQzUz jxTuHj"
        display="block"
      >
        Size of the Google Analytics JavaScript Snippet
      </span>
      <span
        class="sc-kDThTU cwQFAH"
        display="block"
      >
        (Source: 
        <a
          class="sc-jRQAMF hVqlct"
          href="https://www.google-analytics.com/analytics.js"
          rel="noopener noreferrer nofollow"
          target="_blank"
        >
          <span
            class="sc-gsDJrp leAAmy"
          >
            Google
          </span>
        </a>
        )
      </span>
    </div>
  </div>
</div>
`;
